### PR TITLE
Use -Werror again in enable_if_supported

### PR DESCRIPTION
**Pull request by masterleinad**
Reverts #15532. Let's see if there is any other reason for not using `-Werror`.

### CODE REVIEW

**Review by kronbichler — APPROVED** [review]
This looks good to me, and at least the OSX platform seems to be happy. Let's wait for additional opinions.

File: cmake/macros/macro_enable_if_supported.cmake [hunk 1]
```diff
@@ -22,8 +22,15 @@
 #
 
 macro(enable_if_supported _variable _flag)
+  # First check if we can use -Werror
+  CHECK_CXX_COMPILER_FLAG("-Werror" DEAL_II_HAVE_FLAG_werror)
```
Comment (tamiko) [inline cmake/macros/macro_enable_if_supported.cmake:26]:
```suggestion
  CHECK_CXX_COMPILER_FLAG("-Werror" DEAL_II_HAVE_FLAG_Werror)
```

File: cmake/macros/macro_enable_if_supported.cmake [hunk 2]
```diff
@@ -22,8 +22,15 @@
...
+
   string(STRIP "${_flag}" _flag_stripped)
 
+  if(DEAL_II_HAVE_FLAG_werror)
```
Comment (tamiko) [inline cmake/macros/macro_enable_if_supported.cmake:30]:
```suggestion
  if(DEAL_II_HAVE_FLAG_Werror)
```

**Review by tamiko — CHANGES_REQUESTED** [review]

**Review by tamiko — APPROVED** [review]

**Pull request merged**